--- a/debian/rules	2017-03-20 23:54:22.832398302 +0100
+++ b/debian/rules	2017-03-20 23:54:38.456398426 +0100
@@ -20,5 +20,6 @@
 DEB_CONFIGURE_EXTRA_FLAGS += \
 	--libdir=/usr/lib/$(DEB_HOST_MULTIARCH) \
 	--libexecdir=/usr/lib/gvfs \
-	--disable-hal
-
+	--disable-hal \
+	--disable-libsystemd-login \
+	--with-systemduserunitdir=no
--- a/debian/gvfs-common.install	2017-03-21 00:25:18.948413006 +0100
+++ b/debian/gvfs-common.install	2017-03-21 00:25:25.608413059 +0100
@@ -1,4 +1,3 @@
 usr/share/locale
 usr/share/man/man1
 usr/share/man/man7
-usr/lib/tmpfiles.d/gvfsd-fuse-tmpfiles.conf

--- a/debian/changelog	2017-07-30 22:10:03.388699265 +0200
+++ b/debian/changelog	2017-07-30 22:09:52.036699106 +0200
@@ -1,3 +1,10 @@
+gvfs (1.28.1-1ubuntu1ppa1) xenial; urgency=medium
+
+  * Non-maintainer upload.
+  * Rip out systemd.
+
+ -- gdm85 <gdm85@users.noreply.github.com>  Sun, 30 Jul 2017 22:09:15 +0200
+
 gvfs (1.28.1-1ubuntu1) xenial; urgency=medium
 
   * Merge with Debian unstable (LP: #1570251). Remaining Ubuntu changes:
